### DISCUSSION TITLE
Getting the following error: "x509: certificate signed by unknown authority" on WSL

### DISCUSSION THREAD

**Discussion by FredericoCoelhoNunes** [Snowflake-Labs/terraform-provider-snowflake#585] [Q&A]
Hi everyone,

I have been trying to use this provider on  WSL (Ubuntu), and I am getting the x509 error when doing `terraform init`.
Here is the full error trace:
```
Error: Failed to install provider
 │
 │ Error while installing chanzuckerberg/snowflake v0.25.1: could not query provider registry for registry.terraform.io/chanzuckerberg/snowflake: failed to retrieve authentication checksums for provider: the request failed after 2
 │ attempts, please try again later: Get "https://github.com/chanzuckerberg/terraform-provider-snowflake/releases/download/v0.25.1/terraform-provider-snowflake_0.25.1_SHA256SUMS": x509: certificate signed by unknown authority
``` 

I haven't been successful in finding a solution so far. Has anyone come across this error in the past?
Thanks!

**Comment by danu165** [comment]
I'm running into the same error. Did you happen to resolve this?

**Comment by FredericoCoelhoNunes** [comment]
Hi @danu165 , I resolved it by manually installing the provider, and then telling terraform to look for the provider in the folder where I installed it - thus bypassing the querying of the provider registry.
I don't exactly remember the commands I used, but I believe you should be able to find the procedure to manually install providers in the official terraform documentation. I think it's a bit tricky due to having to set the corrects paths.

**Reply by danu165** [reply]
I am actually manually installing the provider, but I receive the same error later after init. Strangely, when I install it manually I get a message saying that it is "unauthenticated":
```
Initializing provider plugins...
--
121 | - Finding latest version of chanzuckerberg/snowflake...
122 | - Installing chanzuckerberg/snowflake v0.25.34...
123 | - Installed chanzuckerberg/snowflake v0.25.34 (unauthenticated)
```

Terraform init succeeds, but then terraform apply gives me the x509 message for snowflake. Did you happen to run into anything like this?
```
Error: error current_account: Post "https://XXX.us-east-1.snowflakecomputing.com:443/session/v1/login-request?requestId=XXX&request_guid=XXX&roleName=XXX": x509: certificate signed by unknown authority
```
